neural/run_workflow.py

```diff
@@ -614,3 +631,158 @@ def test_run(q_docpartitions, q_fold_config_map, bertmodel, train_val_dir, test_
+                             pdropout=docencoder_config['pdropout'],
+                             config=docencoder_config['generic_config'],
+                             gpu_index=gpu_index)
+
+    # doc category scorer
+    num_labels = len(class_weights)
+    doc_categ_scorer = DocCategScorer(docscorer_config['input_dim'], num_labels)
+
+    # define optimizer and group parameters
+    models = [(sent_encoder, 'sent_encoder'), (doc_encoder, 'doc_encoder'), (doc_categ_scorer, 'doc_categ_scorer')]
+
+    if(state_dict_dir):  # load state dictionary of saved models
+        for m, m_name in models:
+            m.load_state_dict(torch.load(os.path.join(state_dict_dir, '{}.pkl'.format(m_name)), map_location=device))
+
+    # update models fdtype and move to device
+    for m, m_name in models:
+        m.type(fdtype).to(device)
+
+    # store attention weights for validation and test set
+    docid_attnweights_map = {dsettype: {} for dsettype in data_loaders if dsettype in {'validation', 'test'}}
+    # store sentences' attention weights
+
+    if(os.path.isfile(os.path.join(sents_embed_dir, 'bert_proc_docs.pkl'))):
+        bert_proc_docs = ReaderWriter.read_data(os.path.join(sents_embed_dir, 'bert_proc_docs.pkl'))
+    else:
+        bert_proc_docs = {}
+
+    print("device: {} | question: {} | fold_num: {} | dsettype: {} | pid: {}"
+          "".format(device, options.get('question'), fold_num, dsettype, pid))
+    pred_class = []
+
+    data_loader = data_loaders[dsettype]
+
+    for m, m_name in models:
+        m.eval()
+
+    sample_counter = 0
+    for i_batch, samples_batch in enumerate(data_loader):
+
+        logprob_scores = []
+
+        docs_batch, docs_len, docs_sents_len, docs_attn_mask, docs_labels, docs_id = samples_batch
+
+        docs_batch = docs_batch.to(device)
+        docs_attn_mask = docs_attn_mask.to(device)
+        docs_sents_len = docs_sents_len.type(torch.int64).numpy()  # to feed this in RNN
+        docs_labels = docs_labels.type(torch.int64).to(device)
+
+        with torch.set_grad_enabled(dsettype == 'train'):
+            # print("number of examples in batch:", docs_batch.size(0))
+            num_docs_perbatch = docs_batch.size(0)
+            for doc_indx in range(num_docs_perbatch):
+                # print('doc_indx', doc_indx)
+                doc_id = docs_id[doc_indx].item()
+                if(doc_id in bert_proc_docs):
+                    # due to GPU limit
+                    embed_sents = torch.load(bert_proc_docs[doc_id], map_location=device)
+                else:
+                    embed_sents = bert_encoder(docs_batch[doc_indx], docs_attn_mask[doc_indx],
+                                               docs_len[doc_indx].item())
+                    # add embedding to dict
+                    embed_fpath = os.path.join(sents_embed_dir, '{}.pkl'.format(doc_id))
+                    ReaderWriter.dump_data(embed_sents, embed_fpath)
```

Comment:
I think here we have the same issue of last PR to harmonize the `reading` and `dumping` using `ReaderWriter.read_tensor` and `ReaderWriter.dump_tensor`